src/compiler/checker.ts

```diff
@@ -254,9 +254,14 @@ module ts {
             return symbolLinks[symbol.id] || (symbolLinks[symbol.id] = {});
         }
 
-        function getNodeLinks(node: Node): NodeLinks {
+        function assignNodeId(node: Node): number {

```

Comment:
I honestly think this could just be a global function. It seems unnecessary to keep in in the checker, and then also have to expose it somewhoe.